rye/src/cli/init.rs

```
@@ -291,3 +393,244 @@ pub fn execute(cmd: Args) -> Result<(), Error> {
+    }
+    Ok(())
+}
+
+fn import_setup_cfg(
+    metadata: &mut Metadata,
+    requirements: &mut BTreeMap<String, String>,
+    path: impl AsRef<Path>,
+) -> Result<(), Error> {
+    let mut ini = Ini::new();
+    ini.set_multiline(true);
+    let config = ini.load(path).map_err(|msg| anyhow::anyhow!(msg))?;
+    if let Some(section) = config.get("metadata") {
+        if metadata.name.is_none() {
+            if let Some(Some(name)) = section.get("name") {
+                metadata.name = Some(name.to_string());
+            }
+        }
+        if metadata.version.is_none() {
+            if let Some(Some(version)) = section.get("version") {
+                metadata.version = Some(version.to_string());
+            }
+        }
+        if metadata.description.is_none() {
+            if let Some(Some(description)) = section.get("description") {
+                metadata.description = Some(description.to_string());
+            }
+        }
+        if metadata.author.is_none() {
+            if let Some(Some(author)) = section.get("author") {
+                let email = match section.get("author_email") {
+                    Some(Some(it)) => it,
+                    _ => "",
+                };
+                metadata.author = Some((author.to_string(), email.to_string()));
+            }
+        }
+        if metadata.license.is_none() {
+            if let Some(Some(license)) = section.get("license") {
+                metadata.license = Some(license.to_string());
+            }
+        }
+    }
+    if let Some(section) = config.get("options") {
+        if metadata.requires_python.is_none() {
+            if let Some(Some(python_requires)) = section.get("python_requires") {
+                metadata.requires_python = Some(python_requires.to_string());
+            }
+        }
+        if let Some(Some(reqs)) = section.get("install_requires") {
+            reqs.lines()
+                .filter_map(|x| Requirement::from_str(x).ok())
```

This will silently fail